app/controllers/concerns/verify_setup_completed.rb

```diff
@@ -28,7 +28,11 @@ module VerifySetupCompleted
   end
 
   def setup_completed?
-    MANDATORY_CONFIGS.all? { |config| SiteConfig.public_send(config).present? }
+    missing_configs.empty?
```

Comment:
This way we're going to iterate all the `MANDATORY_CONFIGS` twice if the setup hasn't been completed yet, once for this check, then another time for generating the message. I wonder if there's a better way to do this, e.g. getting rid of `setup_completed?` renaming `missing_configs` to `missing_mandatory_configs` and changing `verify_setup_completed` like this:

```ruby
missing_configs = missing_mandatory_configs
return if config_path? || missing_configs.empty? || SiteConfig.waiting_on_first_user

link = ...
 missing_configs_text = missing_configs.map...
```